web/rest/user/config/api/raw/provider.yml

```
@@ -111,12 +115,56 @@ Ivoz\Provider\Domain\Model\Voicemail\Voicemail:
         tags:
         - My
         parameters: []
+
+    get_my_voicemails:
+      method: 'GET'
+      path: '/my/voicemails'
+      route_name: 'get_my_voicemails'
+      operation_normalization_context: !php/const Ivoz\Provider\Domain\Model\Voicemail\VoicemailDto::CONTEXT_MY_VOICEMAILS
+      normalization_context:
+        groups: [
+          !php/const Ivoz\Provider\Domain\Model\Voicemail\VoicemailDto::CONTEXT_MY_VOICEMAILS
+        ]
+      swagger_context:
+        produces:
+        - 'application/json'
+        - 'application/ld+json'
+        tags:
+        - My
+        parameters: []
+  properties:
+    name:
+      attributes:
+        swagger_context:
+          readOnly: true
+    email:
+      attributes:
+        swagger_context:
+          readOnly: true
   attributes:
     access_control: '"ROLE_COMPANY_USER" in roles'
+    read_access_control:
+      ROLE_COMPANY_USER:
+        user:
+          eq: "user.getId()"
     swagger_context:
       required:
       - company
 
+Ivoz\Provider\Domain\Model\VoicemailMessage\VoicemailMessage:
+  itemOperations:
+    get: ~
+    delete: ~
+  collectionOperations:
+    get: ~
+
+  attributes:
+    access_control: '"ROLE_COMPANY_USER" in roles'
+    read_access_control:
+      ROLE_COMPANY_USER:
+        voicemail:
+          in: "voicemailRepository.getVoicemailsForUser(user)"
```

We just need an array of ids here. Add another repository method that returns `array<array-key, int>`. 
```suggestion
          in: "voicemailRepository.getVoicemailIdsByUser(user)"
```
This method could make use of getVoicemailIByUser internally